code_conventions.md

```diff
@@ -157,6 +181,40 @@ for (int i = 0; i < len; i++)
  * When calling the index `[]` operator, don't insert a space before the `[`.
 
 
+Static Typing (Python)
+--------
+Normally, Python is using dynamic typing. For example:
+~~~~~~~~~~~~~~~{.py}
+def fib(n):
+    a, b = 0, 1
+    while a < n:
+        yield a
+        a, b = b, a + b
+
+x = fib(5)
+y = fib(2.5)
+~~~~~~~~~~~~~~~
+which defines the a dynamically typed function to return the Fibonacci number.
+This will be fine for x, but what will the result be for y?
+
+Adding more strict (static) typing can prevent these kinds of errors using tools like mypy (http://mypy-lang.org/):
+~~~~~~~~~~~~~~~{.py}
+from typing import Iterator
+
+def fib(n: int) -> Iterator[int]:
+    a, b = 0, 1
+    while a < n:
+        yield a
+        a, b = b, a + b
+
+x = fib(5)
+y = fib(2.5)
+~~~~~~~~~~~~~~~
+In this case, mypy will inform that the call using the floating point value is not valid:
+__mypytest.py:10: error: Argument 1 to "fib" has incompatible type "float"; expected "int"__
+
+This will help catching unwanted and unexpected bugs. This is not obligate but highly preferred and encouraged.
```

Comment:
There are a few spelling errors in these sentences. Probably better to do something like "This will help catch errors and unexpected bugs. This is not mandatory but strongly preferred and encouraged."

Reply:
no bug is ever expected, or if it if it's a feature, not a bug.